torch/_inductor/pattern_matcher.py

```diff
@@ -692,13 +693,14 @@ def check_fn(match: Match):
                 if grad and is_integer_dtype(args[i].dtype):
                     return False
 
-                args[i] = torch.empty_strided(
-                    args[i].size(),
-                    args[i].stride(),
-                    dtype=args[i].dtype,
-                    device=args[i].device,
-                    requires_grad=grad,
-                )
+                with torch._dynamo.utils.detect_fake_mode(args):
```

Comment:
Wonder what does this mean?

Reply:
it's the api to grab FakeTensorMode either from the inputs, from the currently enabled FakeTensorMode on the stack, or from the current TracingContext, which includes things like guards/etc that is generated from dynamo. It makes it so that the empty_strided tensor will be a fake tensor, so that it doesnt error when trying to instantiate a tensor with symbolic shapes as size. 

FakeTensors correspond to a single fake tensor mode. It is not intended to use FakeTensors from different modes, because the Mode has things like cached storage conversions for correctness. So we reuse the existing mode instead of creating a new one.